File: May-Leetcoding-Challenge/19. KthSmallestElementInaBST/KthSmallestElementInaBST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int kthSmallest(TreeNode root, int k) 
    {
        List<Integer> a = new ArrayList<>();
        List<Integer> b = new ArrayList<>();
        
        b = inorder(root, a);
        return b.get(k-1);
    }
    public List<Integer> inorder(TreeNode root, List<Integer> a)
        {
            if(root == null)
                return a;
            inorder(root.left, a);
            a.add(root.val);
            inorder(root.right,a);
            return a;
        }
}

File: May-Leetcoding-Challenge/20. KthSmallestElementInaBST/KthSmallestElementInaBST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def kthSmallest(self, root: TreeNode, k: int) -> int:
        if not root:
            return
        
        a = []    
        def inorder(root):
            if not root:
                return 
            inorder(root.left)
            a.append(root.val)
            inorder(root.right)
            return a 
        
        b = inorder(root)
        #print(b)
        return b[k-1]
    
